plugins/org.yakindu.sct.simulation.core.sexec/src/org/yakindu/sct/simulation/core/sexec/interpreter/DefaultExecutionFlowInterpreter.xtend

```diff
@@ -205,11 +207,13 @@ class DefaultExecutionFlowInterpreter implements IExecutionFlowInterpreter, IEve
 			
 		}
 	}
-
+	
 	def superStepLoop(()=>void microStep) {
 		do {
 			stateVectorChanged = false
 			microStep.apply
+			// give the highlighter a chance to do its work
+			Thread.sleep(70)
```

Comment:
We can not sleep 70ms within each superstep cycle !!

Reply:
Replaced Thread.sleep by an explicit check for Thread.isInterrupted(). Now the user can click on terminate, but it might still take some seconds until it is performed.